sematic/resolvers/worker.py

```diff
@@ -149,7 +150,18 @@ def main(run_id: str, resolve: bool):
         logger.error("Run failed:")
         logger.error("%s: %s", e.__class__.__name__, e)
 
-        _fail_run(run)
+        if resolve:
+            # refresh the run from the DB in case it was updated from
+            # its worker job
+            root_run = api_client.get_run(run_id)
+            if not FutureState[root_run.future_state].is_terminal():  # type: ignore
```

Comment:
Again, not strictly required as a part of this PR, but I found it helpful to have during dev

Reply:
There is an `if resolve` on line 166, can we combine?

Reply:
no, because this has to come from the re-loaded root run, which we don't have to get unless `resolve` is True. So it must be:
resolve is True, therefore re-pull run. Use pulled run's status to determine whether failure is needed.